pkg/client/unversioned/conditions.go

```diff
@@ -54,11 +54,15 @@ func JobHasDesiredParallelism(c ExtensionsInterface, job *extensions.Job) wait.C
 		}
 
 		// desired parallelism can be either the exact number, in which case return immediately
-		if job.Status.Active == *job.Spec.Parallelism {
+		if job.Spec.Parallelism != nil && job.Status.Active == *job.Spec.Parallelism {
```

Comment:
Parallelism will never be nil, since it's defaulted if not present. No need to change that.